### PR TITLE
cord-api-130

### PR DESCRIPTION
**Pull request by gowthamrodda**
this has a user attached to file with createdby relationship.

### CODE REVIEW

File: src/components/file/file.service.ts [hunk 1]
```diff
@@ -103,25 +107,50 @@ export class FileService {
...
+      let user: User | null;
+      const userSession: ISession = await this.authService.decodeAndVerifyToken(session.token);
+      user = (userSession.userId) ? await this.userService.readOne(userSession.userId, session) : null;
+
       const result = await this.db
         .query()
         .raw(
           `
-        MATCH (token:Token {active: true, value: $token})
+        MATCH
```
Comment (candleshine) [inline src/components/file/file.service.ts]: There is a create abstraction in property-updater service see education service for how it is used.

File: src/components/file/file.service.ts [hunk 2]
```diff
@@ -103,25 +107,50 @@ export class FileService {
...
+        <-[:token {active: true}]-
+        (requestingUser:User {
+          active: true,
+          id: $requestingUserId
+        })
         CREATE
-            (file:FileNode { id: $id, type: $type, name: $name })
+          (file:FileNode {
+            id: $id,
+            type: $type,
```
Comment (candleshine) [inline src/components/file/file.service.ts]: these values should not be properties on the node.  If you look at the database schema.jpg, you will see that these are property nodes.  The create function in property-updater service will do this for you.  you just have to pass in the key/values into the function

File: test/file.e2e-spec.ts [hunk 3]
```diff
@@ -1,25 +1,75 @@
...
     app = await createTestApp();
   });
 
-  it('read file node by id', async () => {
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('Create FileNode', async () => {
     const token = await createSession(app);
```
Comment (candleshine) [inline test/file.e2e-spec.ts]: I re ordered these so Create happens before Read.  token is not used by this function

File: src/components/file/file.service.ts [hunk 4]
```diff
@@ -103,25 +107,50 @@ export class FileService {
   ): Promise<File> {
     try {
       // TODO find a better way to check if object exists in s3 and move
+      let user: User | null;
+      const userSession: ISession = await this.authService.decodeAndVerifyToken(session.token);
+      user = (userSession.userId) ? await this.userService.readOne(userSession.userId, session) : null;
```
[the remark this answers is not in the document]
Comment (gowthamrodda) [reply src/components/file/file.service.ts]: to pull user id from provided token

File: src/components/file/file.service.ts (hunk 1 above)
Comment (gowthamrodda) [reply src/components/file/file.service.ts]: done

**Pull request merged**